bls/src/main/swig/bls-signatures.i

```diff
@@ -0,0 +1,243 @@
+%ignore operator*(const G2Element &a, const bn_t &k);
+%ignore operator*(const bn_t &k, const G2Element &a);
+%ignore operator*(const PrivateKey& a, const bn_t& k);
+%ignore operator*(const bn_t& k, const PrivateKey& a);
+
+// unsigned char [] to byte []
+%typemap(jni) (unsigned char *) "jbyteArray"
+%typemap(jtype) (unsigned char *) "byte[]"
+%typemap(jstype) (unsigned char *) "byte[]"
+%typemap(in) (unsigned char *) {
+  $1 = (unsigned char *) JCALL2(GetByteArrayElements, jenv, $input, 0);
+}
+
+%typemap(argout) (unsigned char *) {
+  JCALL3(ReleaseByteArrayElements, jenv, $input, (jbyte *) $1, 0);
+}
+
+%typemap(javain) (unsigned char *) "$javainput"
+
+%typemap(freearg) (unsigned char *) ""
+
+
+// vector<uint8_t> to byte[]
+%typemap(jni) std::vector<uint8_t> "jbyteArray"
+%typemap(jtype) std::vector<uint8_t> "byte[]"
+%typemap(jstype) std::vector<uint8_t> "byte[]"
+%typemap(javain) std::vector<uint8_t> "$javainput"
+%typemap(javaout) std::vector<uint8_t> { return $jnicall; }
+
+%typemap(in) std::vector<uint8_t> (std::vector<uint8_t> vec) {
+  if (!$input) {
+    SWIG_JavaThrowException(jenv, SWIG_JavaNullPointerException, "null array");
+    return $null;
+  }
+  const jsize sz = jenv->GetArrayLength($input);
+  jbyte* const jarr = jenv->GetByteArrayElements($input, 0);
+  if (!jarr) return $null;
+  vec.assign(jarr, jarr+sz);
+  jenv->ReleaseByteArrayElements($input, jarr, JNI_ABORT);
+  $1 = &vec;
+}
+
+%typemap(out) std::vector<uint8_t> {
+  const jsize sz = $1.size();
+  $result = jenv->NewByteArray(sz);
+  jenv->SetByteArrayRegion($result, 0, sz, reinterpret_cast<jbyte *>($1.data()));
+}
+
+%apply std::vector<uint8_t> { const std::vector<uint8_t> & };
+
+// ignore functions with vector<uint8_t>
+namespace bls {
+    %ignore PrivateKey::FromByteVector;
+    %ignore G1Element::FromByteVector;
+    %ignore G2Element::FromByteVector;
+    %ignore GTElement::FromByteVector;
+    %ignore G1Element::FromMessage(const std::vector<uint8_t>& message, const uint8_t* dst, int dst_len, bool fLegacy = false);
+    %ignore G2Element::FromMessage(const std::vector<uint8_t>& message, const uint8_t* dst, int dst_len, bool fLegacy = false);
+    %ignore HDKeys::KeyGen(const std::vector<uint8_t>& seed);
+    %ignore CoreMPL::KeyGen(const vector<uint8_t>& seed);
+    %ignore CoreMPL::Sign(const PrivateKey &seckey, const vector<uint8_t> &message);
+    %ignore CoreMPL::Verify(const G1Element &pubkey, const vector<uint8_t> &message, const G2Element &signature);
+    %ignore CoreMPL::Verify(const vector<uint8_t> &pubkey, const vector<uint8_t> &message, const vector<uint8_t> &signature);
+    %ignore PopSchemeMPL::PopVerify(const vector<uint8_t> &pubkey, const vector<uint8_t> &proof);
+    %ignore PopSchemeMPL::FastAggregateVerify(const vector<G1Element> &pubkeys, const vector<uint8_t> &message, const G2Element &signature);
+    %ignore AugSchemeMPL::Sign(const PrivateKey &seckey, const vector<uint8_t> &message, const G1Element &prepend_pk);
+}
```

Comment:
The BLS library has many functions that take `Bytes` and `vector<uint8_t>`.  Since both are mapped to `byte[]` that would result in many duplicated functions in the Java wrapper classes, so we will ignore the methods that take `vector<uint8_t>`.